pkg/apis/extensions/types.go

```diff
@@ -437,6 +449,22 @@ type RollingUpdateDaemonSet struct {
 	MaxUnavailable intstr.IntOrString
 }
 
+// Spec to control the desired behavior of a daemon set surging rolling update.
+type SurgingRollingUpdateDaemonSet struct {
```

Comment:
We decided to go with a separate strategy in order to minimize the risk of introducing bugs in the existing paths. Also, bearing in mind that the only use case we have today for maxSurge in daemonsets is self-hosting, we decided to go down the separate strategy path.